mbed_devices/mbed_devices.py

```diff
@@ -25,13 +28,19 @@ def _build_device(candidate: CandidateDevice) -> Device:
     )
 
 
+class NoTargetForCandidate(Exception):
+    """Raised when target cannot be determined for a candidate."""
+
+    pass
+
+
 def _get_mbed_target_for_candidate(candidate: CandidateDevice) -> Optional[MbedTarget]:
```

Comment:
Does it still return `Optional[MbedTarget]`? Would be nice to not have to use Optional at all and remove it from the imports.

Reply:
Nice catch! Type hinting is sooo goood. 🤢

Reply:
https://github.com/ARMmbed/mbed-devices/pull/40/commits/c5a9cf70cdb02af01b4bc868ac75320c33746ec4